src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java

```diff
@@ -0,0 +1,67 @@
+package mate.academy.dao.impl;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+    @Override
+    public MovieSession add(MovieSession movieSession) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t insert movie session: " + movieSession, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+    
+    @Override
+    public Optional<MovieSession> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(MovieSession.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can`t get movie session by id: " + id, e);
+        }
+    }
+    
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        LocalDateTime minTime = date.atTime(0,0,0);
+        LocalDateTime maxTime = date.atTime(23,59,59);
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> query = session.createQuery("from MovieSession ms "
+                    + "join fetch ms.cinemaHall "
+                    + "join fetch ms.movie where ms.movie.id = :id and "
+                    + "ms.showTime between :min and :max", MovieSession.class);
```

Comment:
```suggestion
            Query<MovieSession> query = session.createQuery("from MovieSession ms "
                    where ms.movie.id = :id and "
                    + "ms.showTime between :min and :max", MovieSession.class);
```
'fetch' is redundant here, because in every *toOne relation it is 'EAGER' by def